src/funcs.rs

```diff
@@ -187,6 +196,77 @@ fn substring(args: &Vec<data::Value>) -> Result<data::Value, EvalError> {
     }
 }
 
+fn to_lower_case(s: &str) -> Result<data::Value, EvalError> {
+    Ok(data::Value::from_string(s.to_lowercase()))
+}
+
+fn to_upper_case(s: &str) -> Result<data::Value, EvalError> {
+    Ok(data::Value::from_string(s.to_uppercase()))
+}
+
+fn is_null(args: &Vec<data::Value>) -> Result<data::Value, EvalError> {
+    match args.as_slice() {
+        [arg0] => Ok(data::Value::Bool(*arg0 == data::Value::None)),
+        _ => Err(EvalError::InvalidFunctionArguments {
+            name: "isNull",
+            expected: 1,
+            found: args.len(),
+        }),
+    }
+}
+
+fn is_empty(args: &Vec<data::Value>) -> Result<data::Value, EvalError> {
+    match args.as_slice() {
+        [data::Value::None] => Ok(data::Value::Bool(true)),
+        [arg0] => Ok(data::Value::Bool(arg0.to_string().is_empty())),
```

Comment:
can match on the specific types of arg0 here? More explicit about how emptiness works, doesn't have a dependency on the to_string impl, and will also save an allocation since we can check the length of the string directly